### PR TITLE
Adapt abstracts CSV and XLSX export

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: indico/modules/events/abstracts/controllers/management.py [hunk 1]
```diff
@@ -279,3 +281,29 @@ def _process(self):
...
+        self.headers, self.rows = generate_spreadsheet_from_abstracts(self.abstracts,
+                                                                      self.export_config['static_item_ids'],
+                                                                      self.export_config['dynamic_items'])
+
+
+class RHAbstractsExportCSV(RHAbstractsExportBase):
+    """Export list of abstracts to CSV"""
+
+    def _process(self):
+        RHAbstractsExportBase._process(self)
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/controllers/management.py]: Wouldn't it be cleaner to have a differently named methods for this which returns those headers and rows?

`return send_csv('abstracts.csv', *self._generate_spreadsheet())`

File: indico/modules/events/abstracts/util.py [hunk 2]
```diff
@@ -227,6 +229,52 @@ def build_default_email_template(event, tpl_type):
...
+def generate_spreadsheet_from_abstracts(abstracts, static_item_ids, dynamic_items):
+    """Generates a spreadsheet data from a given abstract list.
+
+    :param abstracts: The list of abstracts to include in the file
+    :param static_item_ids: The abstract properties to be used as columns
+    :param dynamic_items: Contribution fields as extra columns
+    """
+    field_names = ['ID', 'Title']
+    static_item_mapping = OrderedDict([
+        ('state', ('State', lambda x: x.state.title)),
```
Comment (OmeGak) [inline indico/modules/events/abstracts/util.py]: `attrgetter('state.title')`

File: indico/modules/events/abstracts/util.py [hunk 3]
```diff
@@ -227,6 +229,52 @@ def build_default_email_template(event, tpl_type):
...
+                                   lambda x: x.accepted_contrib_type.name if x.accepted_contrib_type else None)),
+        ('submitted_contrib_type', ('Submitted type',
+                                    lambda x: x.submitted_contrib_type.name if x.submitted_contrib_type else None)),
+        ('score', ('Score', lambda x: round(x.score, 1))),
+        ('submitted_dt', ('Submission date', lambda x: to_unicode(format_datetime(x.submitted_dt)))),
+        ('modified_dt', ('Modification date', lambda x: (to_unicode(format_datetime(x.modified_dt)) if x.modified_dt
+                                                         else '')))
+    ])
+    for item in dynamic_items:
+        field_names.append(unique_col(item.title, item.id))
```
Comment (pferreir) [inline indico/modules/events/abstracts/util.py]:
```python
field_names += [unique_col(item.title, item.id) for item in dynamic_items]
```
?

File: indico/modules/events/abstracts/util.py (hunk 2 above)
Comment (nop33) [inline indico/modules/events/abstracts/util.py]: Is there any benefit over using this instead of a lambda?

Comment (OmeGak) [inline indico/modules/events/abstracts/util.py]: Performance, as `attrgetter` is implemented in C. Not that it matters much in this particular case.

Comment (nop33) [inline indico/modules/events/abstracts/util.py]: I see. I'd like to keep it like this, makes the code more consistent in some sense. But not a strong argument.

**Pull request merged**